Marc revisions (#21)
Nikoleta-v3/meta-analysis-of-prisoners-dilemma-tournaments: paper/main.tex

\end{figure}

The distributions of the SSE across the tournament types suggest that successful
strategies exhibit some extortionate behaviour, but not constantly.
ZDs are a set of strategies that are envious as they try to exploit their
opponents. The winners of the tournaments considered in this work are
envious, but not as much as many ZDs. This highlights why TFT's early tournament
success fails to generalize -- it never attempts to defect against a cooperating
or exploitable opponent (e.g. Alternator). Moreover, many of the strategies in
the library will not tolerate exploitation attempts. A clever strategy can
achieve mutual cooperation with stronger strategies while also being able to

> exploit weaker strategies. This is why ZD strategies fail to appear in the top

Reviewer: ```suggestion
exploit weaker strategies. This is why ZDs fail to appear in the top
```

ranks -- they try to exploit all opponents and cannot actively adapt back to
mutual cooperation against stronger strategies, which requires more depth of memory.

> \footnote{Note that ZD strategies also tend to perform poorly in population games for

Reviewer: ```suggestion
\footnote{Note that ZDs also tend to perform poorly in population games for
```

> a similar reason: they attempt to exploit other players using ZD strategies, failing to

Reviewer: ```suggestion
a similar reason: they attempt to exploit other players using ZDs, failing to
```

\end{figure}

A third method that evaluates the importance of the features in
Table~\ref{table:manual_features} using clustering and random forests can be found in
the Appendix~\ref{app:clustering}. The results uphold the outcomes of the
correlation and multivariate regression methods. It also evaluates the effects
of the classifiers stochastic, make use of game, and make use of length which
have not been evaluated by the methods above because there are binary variables.
The results imply that they have no significant effect on a strategy's
performance.

\section{Discussion}\label{section:conclusion}

This manuscript has explored the performance of \numberofstrategies strategies
of the Iterated Prisoner's Dilemma in a large number of computer tournaments.
We analyzed and extracted the salient features of the best performing strategies
across various tournament types, casting the results in terms of Axelrod's
original suggested features of good IPD strategies. Moreover, our results

> shed light on the historic performance of TFT, zero determinant strategies,

Reviewer: ```suggestion
shed light on the historic performance of TFT, zero-determinant strategies,
```

> and memory-one strategies generally. Strategies need to match their play

Reviewer: ```suggestion
and memory one strategies generally. Strategies need to match their play
```

to the cooperativeness of the tournament population and do so in a way that
prevents or minimizes exploitation. Overall we see that complex or clever
strategies can be effective, whether trained against a corpus of possible
opponents or purposely designed to mitigate the impact of noise such as
in the strategy DBS. Further, we showed that while the type of exploitation

> attempted by ZD strategies is not typically effective in tournaments, more

Reviewer: ```suggestion
attempted by ZDs is not typically effective in tournaments, more
```

sophisticated strategies capable of selectively exploiting weaker opponents
while mutually cooperating with stronger opponents can be highly successful.
This fact was also indicated numerically by the importance of the strategy
feature SSE in the analysis of strategy features. These results highlight
a central idea in evolutionary game theory in this context: the fitness
landscape is a function of the population (where fitness in this case is
tournament performance). While that may seem obvious now, it shows why
historical tournament results on small or arbitrary populations of strategies
have so often failed to produce generalizable results.

Highly noisy or tournaments with short matches favor less cooperative strategies.
These environments mitigate the value of being nice. Uncertainty enables
exploitation, reducing the ability of maintaining or enforcing mutual cooperation,
while triggering grudging strategies to switch from typically cooperating to
typically defecting. Accordingly we find that in noisy tournaments

> that the best performing players cooperate a lower rate than the tournament

Reviewer: ```suggestion
typically defecting. Accordingly, we find that in noisy tournaments
the best performing players cooperate a lower rate than the tournament
```
